### PR TITLE
Add Mark as read feature in inbox

### PR DESCRIPTION
**Pull request by uwaiszaki**

### CODE REVIEW

**Comment by uwaiszaki** [comment]
@xamanu To add Mark as read should I use a patch request or a simple get request(getMessage) as on using get request it is marking the message as read.

**Comment by uwaiszaki** [comment]
@xamanu @ethan-nelson @ramyaragupathy @smit1678  Now messages are grouped by their "read" status.
How should I add sorting according to date?

**Comment by uwaiszaki** [comment]
I have added a patch request in the MessageAPI to change read status without visiting the message page.

**Comment by xamanu** [comment]
@uwaiszaki, may you please reroll this on the lastest `develop` and I suggest to use #1423 as a base, too.

**Comment by uwaiszaki** [comment]
@xamanu this will add "mark as read" feature, in continuation of #1423.

**Comment by xamanu** [comment]
Can you please rebase, @uwaiszaki?

File: server/api/messaging/message_apis.py [hunk 1]
```diff
@@ -238,6 +238,50 @@ def delete(self, message_id):
             current_app.logger.critical(error_msg)
             return {"error": error_msg}, 500
 
+    @tm.pm_only(False)
+    @token_auth.login_required
+    def patch(self, message_id):
```
Comment (uwaiszaki) [inline server/api/messaging/message_apis.py]: @xamanu I have used a patch request here.

**Comment by uwaiszaki** [comment]
@xamanu I have rebased the pr with develop branch, please give your reviews.

**Comment by xamanu** [comment]
Just reviewed. Works well, but I'm not sure whether this is the desired behaviour:

image

I would prefer to reuse the checkbox on the beginning and out of the "Delete all selected" I would combine it another action "Mark as read all selected messages"

path2-53-7-8-2

@willemarcel @russdeffner What do you think?

**Comment by willemarcel** [comment]
I agree with you, @xamanu . We could make the unread messages bold to distinguish it from the read ones.

**Comment by willemarcel** [comment]
Other thing: could we put the filter line outside the messages table?

**Comment by uwaiszaki** [comment]
@willemarcel @xamanu I think it would be really confusing for the user if the checkbox is not marked after marking it as read as the same checkbox will be used for deleting multiple messages. Can we show the read messages on another page? ex. like the second page in the Gmail inbox

**Comment by uwaiszaki** [comment]
This currently adds API and services to mark/unmark multiple messages as read. Placement of delete and mark buttons as a settings menu is remaining.

**Comment by xamanu** [comment]
I don't think the read status should be indicated by a checked checkbox. Also the redesign separates the status indicator (line on the left of each message) with the function to mark it as read.

**Comment by xamanu** [comment]
As per conversation with @uwaiszaki targeting this for the redesign and shall be implemented with React.

**Pull request closed without merging**